### DISCUSSION TITLE
Why `experimental.renderBuiltUrl` not called for every asset/chunk?

### DISCUSSION THREAD

**Discussion by a1mersnow** [vitejs/vite-plugin-vue#113] [Q&A]
The [Doc](https://vitejs.dev/guide/build.html#advanced-base-options).

I saw the option in `vitejs/vite-plugin-vue` repo, the `playground/ssr-vue/vite.config.js`:

```js
  experimental: {
    renderBuiltUrl(filename, { hostType, type, ssr }) {
      console.log(filename)
      if (ssr && type === 'asset' && hostType === 'js') {
        return {
          runtime: `__ssr_vue_processAssetPath(${JSON.stringify(filename)})`,
        }
      }
    },
  }
```
As you can see, I add a `console`.
When I run `pnpm build:client`, the `renderBuiltUrl` is called per assets/chunk.
However, when I run `pnpm build:server`, the `renderBuiltUrl` is only called for a part of them.

I want to know why.

**Comment by Zack4DEV** [comment]
@a1mersnow  

try to add assign Specific Values for { hostType, type, ssr }
{ hostId, hostType, type }: { hostId: string, hostType:  'js' | 'css' | 'html', type: 'public' | 'asset' }
